ambry-api/src/main/java/com/github/ambry/rest/RequestPath.java

```diff
@@ -55,6 +60,13 @@ public static RequestPath parse(RestRequest restRequest, List<String> prefixesTo
       // If blob id is specified in the rest request header and it has cluster name prefix, we remove its prefix, subResource
       // and use this id (pure blob id with no prefix or subResource) to update "x-ambry-blob-id" header. The purpose is
       // to ensure IdConverter and IdSigningService can receive a valid blob id and correctly identify it.
+      try {
+        blobIdHeader = URLDecoder.decode(blobIdHeader, StandardCharsets.UTF_8.name());
+      } catch (UnsupportedEncodingException e) {
+        throw new RestServiceException(
+            "Fail to decode the blobIdHeader value: " + blobIdHeader + " for signed named blob: " + e.getMessage(),
```

Comment:
Can you please update the error message? the blobId is not just for signed named blob.

Reply:
Please add some test cases for this code.